OOP-functions/oop/app.js

```diff
@@ -0,0 +1,426 @@
+            element.style.display = "none";
+            if (items[i].match(pattern)){
+                element.style.display = "flex";
+            }     
+        }
+        for (i in itemsComplete) {
+            let element = document.getElementById("Complete_" + i);
+            element.style.display = "none";
+            if (itemsComplete[i].match(pattern)){
+                element.style.display = "flex";
+            }     
+        }
+    }
+
+    addItem = () => {
+        const newItemInput = document.getElementById('NewItemInput');
+
+        this.setState({
+            items: [...this.state.items, 'item' + (this.state.items.length + 1)]
+        });
+        const screenlock = document.getElementById("screenlock");
+        screenlock.style.display = "none";        
+    }
+
+    ApplyItem = () => {
+        const newItemInput = document.getElementById("NewItemInput");
+        this.setState({
+            items: [...this.state.items, newItemInput.value]
+        });
+
+        const screenlock = document.getElementById("ScreenLock");
+        screenlock.style.display = "none"
+    }
+
+    removeItem = (element) => {
+        let states = this.state.items;
+        const parent = element.srcElement.parentElement.id;
+        const label = parent.replace("Task_", "TasksLabel_");
+        const item = document.getElementById(label);
+        const removed = states.splice(states.indexOf(item.innerHTML), 1);
+
+        this.setState({
+            items: [...states]
+        });
+        return removed[0];
+    }
+
+    removeCompletedItem = (element) => {
+        let states = this.state.completeItems;
+        const parent = element.srcElement.parentElement.id;
+        const label = parent.replace("Complete_", "CompleteLabel_");
+        const item = document.getElementById(label);
+        const removed = states.splice(states.indexOf(item.innerHTML), 1);
+
+        this.setState({
+            completeItems: [...states]
+        });
+        return removed[0];
+    }
+
+    CancelAction = () => {
+        const screenlock = document.getElementById("ScreenLock");
+        screenlock.style.display = "none"; 
+    }
+
+
+    NewTask = () => {
+        const newItemButtonApply = document.getElementById('NewItemButtonApply');
+        newItemButtonApply.classList.remove("newitembox__button--enabled");
+        newItemButtonApply.classList.add("newitembox__button--disabled");
+        newItemButtonApply.disabled = true;
+
+        const screenlock = document.getElementById("ScreenLock");
+        screenlock.style.display = "flex"
+        const newItemBox = document.getElementById('NewItemInput');
+        newItemBox.value = '';
+        newItemBox.focus();
+    }
+
+    NewTaskSearch = () => {
+        console.log('test');
+        const newItemBox = document.getElementById('NewItemButtonApply');
+        newItemBox.disabled == false ? newItemBox.onclick.apply() : '';
+    }
+
+    AddRows = (items) => {
```

Comment:
The same point, lets try to create a component for completed/uncompleted rows, create button, header, list of tasks and etc.